File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Polygonator - Menon e Dalla Via/Java/src/Model/Point.java
package Model;

public class Point {
    private double x, y;

    public Point(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    /**
     * Returns the point-slope equation of a straight line
     */
    public String getEquation() {
        return "y " + (y >= 0 ? "- " : "+ ") + String.format("%.2f", y) + " = m(x " + (x >= 0 ? "- " : "+ ") + String.format("%.2f", y) + ")";
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof Point) {
            Point p = (Point) obj;

            return x == p.getX() && y == p.getY();
        }

        return false;
    }

    public boolean smaller(Point p) {
        return x < p.getX() || (x == p.getX() && y < p.getY());
    }

    public boolean greater(Point p) {
        return x > p.getX() || (x == p.getX() && y > p.getY());
    }

    @Override
    public String toString() {
        return "(" + String.format("%.2f", x) + "," + String.format("%.2f", y) + ")";
    }
}

File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Polygonator - Menon e Dalla Via/C++/Model/Point.cpp
#include "Point.h"

Point::Point(double x, double y) : x(x), y(y) {}

double Point::getX() const {
    return x;
}

double Point::getY() const {
    return y;
}

string Point::getEquation() const {
    char sx[10], sy[10];
    sprintf(sx, "%.2f", fabs(x));
    sprintf(sy, "%.2f", fabs(y));
    return string("y ") + (y>=0 ? "- " : "+ ") + sy + " = m(x " + (x>=0 ? "- " : "+ ") + sx + ")";
}

bool Point::operator==(const Point &p) const {
    return x == p.x && y == p.y;
}

bool Point::operator!=(const Point &p) const {
    return !(*this == p);
}

bool Point::operator<(const Point &p) const {
    return x < p.x || (x == p.x && y < p.y);
}

bool Point::operator>(const Point &p) const {
    return x > p.x || (x == p.x && y > p.y);
}

bool Point::operator<=(const Point &p) const {
    return *this == p || *this < p;
}

bool Point::operator>=(const Point &p) const {
    return *this == p || *this > p;
}
